### PR TITLE
Point to the latest URL for the description.

### PR DESCRIPTION
**Pull request by tom24d**
Dear community,

This is a tiny patch though, 
point to 3.x document URL as Python 2.x has been deprecated a long time ago.

Thanks,

### CODE REVIEW

File: doc/reference/glossary.rst [hunk 1]
```diff
@@ -8,7 +8,7 @@ Glossary
    dictionary
       A Python dictionary maps keys to values. Also known as "hashes",
       or "associative arrays" in other programming languages.
-      See https://docs.python.org/2/tutorial/datastructures.html#dictionaries
+      See https://docs.python.org/3/tutorial/datastructures.html#dictionaries
```
Comment (rossbar) [inline doc/reference/glossary.rst:11]:
```suggestion
      See :ref:`the Python tutorial on dictionaries <tut-dictionaries>`
```

**Review by rossbar — COMMENTED** [review]
Thanks @tom24d , very nice catch!

I think it'd be even better if we let intersphinx handle the URL resolution so that we can guarantee the link doesn't go stale in the future - see suggestion below.

**Review by rossbar — APPROVED** [review]
LGTM, thanks @tom24d !

**Comment by MridulS** [comment]
Is it just me or CircleCI isn't building here?

**Comment by rossbar** [comment]
> Is it just me or CircleCI isn't building here?

Not just you - I see it too.

@tom24d is there any chance that you have circleCI enabled on your fork? I know that this sometimes (always?) prevents circleCI from running in the "main" fork.

**Comment by tom24d** [comment]
> is there any chance that you have circleCI enabled on your fork?

No. 
I will reopen this PR to kick circleci.

**Comment by rossbar** [comment]
No worries @tom24d - I think there's just something wrong with circleCI at the moment; nothing related to your PR!

I've built the changes locally and can confirm the link renders correctly. If we want a hosted preview, I'm happy to push them up to gh-pages on my fork - just LMK @MridulS !

**Review by jarrodmillman — APPROVED** [review]

**Pull request merged**